### PR TITLE
docs: Regions and Navigators

### LINKED ISSUE #2159: [Navigation] Uno.Extensions.Navigation.UI..Region Logger not set prior to navigating in MVUX project.
### Current behavior

Use Uno Wizard to setup a MVUX project which allows us to use NavigationView paired with Regions.
With minimum modifications, update the shells.xaml.cs xaml to include the markup below.
Leave most of the app.cs file the same. I removed the secondpage from the project.

A. shell.xaml file should contain two groups of markup. One group is commented out, the other is not. Uncomment and run the "NavigationView" markup. The execution reports an error: ...  Region Logger is null.

B. Comment out the "NavigationView" markup and uncomment the "extended splash screen" markup. Run the "extended splash screen" markup. The Region Logger is set to a valid logger.

### Expected behavior

The expected behavior for both sets of markup is that the Region Logger is properly set to a valid logger and will not throw and exception.

### How to reproduce it (as minimally and precisely as possible)

Use Uno Wizard to setup a MVUX project which allows us to use NavigationView paired with Regions.
With minimum modifications, update the shells.xaml.cs xaml to include the markup below.
Leave most of the app.cs file the same. I removed the secondpage from the project.

A. shell.xaml file should contain two groups of markup. One group is commented out, the other is not. Uncomment and run the "NavigationView" markup. The execution reports an error: ...  Region Logger is null.

B. Comment out the "NavigationView" markup and uncomment the "extended splash screen" markup. Run the "extended splash screen" markup. The Region Logger is set to a valid logger.

Sample markup to test with.

<UserControl x:Class="UnoExtenApp005.Presentation.Shell"
      xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
      xmlns:local="using:UnoExtenApp005.Presentation"
      xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
      xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
      xmlns:muxc="using:Microsoft.UI.Xaml.Controls"  
      xmlns:uen="using:Uno.Extensions.Navigation.UI"
      xmlns:utu="using:Uno.Toolkit.UI"
      xmlns:um="using:Uno.Material"             
      mc:Ignorable="d"
      d:DesignHeight="300"
      d:DesignWidth="400">

  
  
  
  

  <Grid uen:Region.Attached="True" uen:Region.Navigator="Visibility">
    <Grid.RowDefinitions>
      <RowDefinition Height="Auto" />
      <RowDefinition />
    </Grid.RowDefinitions>

    <utu:NavigationBar Content="Main Page"  />

    <muxc:NavigationView Grid.Row="1" x:Name="NavView" uen:Region.Attached="True">
      <muxc:NavigationView.MenuItems>
        <muxc:NavigationViewItem Content="Main" uen:Region.Name="Main" />
      </muxc:NavigationView.MenuItems>

      <muxc:NavigationView.Content>
        <Grid>
          <Grid.RowDefinitions>
            <RowDefinition />
            <RowDefinition Height="Auto" />
          </Grid.RowDefinitions>

          <Grid uen:Region.Attached="True">
            <Grid Visibility="Collapsed" uen:Region.Name="Main">
              <TextBlock Text="Main"
                                  FontSize="24"
                                  HorizontalAlignment="Center"
                                  VerticalAlignment="Center" />
            </Grid>

          </Grid>
        </Grid>
  
      </muxc:NavigationView.Content>
    </muxc:NavigationView>

  </Grid>

  

</UserControl>  

### Workaround

I don't know of a work around.

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

<Project ToolsVersion="15.0">
  <ItemGroup>
    <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="8.0.0" />
    <PackageVersion Include="coverlet.collector" Version="6.0.0" />
    <PackageVersion Include="FluentAssertions" Version="6.12.0" />
    <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
    <PackageVersion Include="NUnit" Version="4.0.1" />
    <PackageVersion Include="NUnit3TestAdapter" Version="4.5.0" />
    <PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
    <PackageVersion Include="Microsoft.Windows.Compatibility" Version="8.0.0" />
    <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.2428" />
    <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231219000" />
    <PackageVersion Include="Serilog.AspNetCore" Version="8.0.1" />
    <PackageVersion Include="SkiaSharp.Skottie" Version="2.88.7" />
    <PackageVersion Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.7" />
    <PackageVersion Include="Swashbuckle.AspNetCore" Version="6.5.0" />
    <PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
    <PackageVersion Include="Uno.Extensions.Core.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Configuration" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Hosting" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Hosting.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Http" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Http.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Http.Refit" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Localization" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Localization.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.7.0" />
    <PackageVersion Include="Uno.Extensions.Logging.Serilog" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Logging.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Navigation" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Navigation.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Navigation.Toolkit.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Reactive" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Reactive.WinUI" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Serialization" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Serialization.Http" Version="4.0.0" />
    <PackageVersion Include="Uno.Extensions.Serialization.Refit" Version="4.0.0" />
    <PackageVersion Include="Uno.Material.WinUI" Version="4.1.0" />
    <PackageVersion Include="Uno.Dsp.Tasks" Version="1.2.8" />
    <PackageVersion Include="Uno.Toolkit.WinUI" Version="5.1.3" />
    <PackageVersion Include="Uno.Toolkit.WinUI.Material" Version="5.1.3" />
    <PackageVersion Include="Uno.Resizetizer" Version="1.2.1" />
    <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
    <PackageVersion Include="Uno.Wasm.Bootstrap" Version="9.0.0-dev.27" />
    <PackageVersion Include="Uno.Wasm.Bootstrap.DevServer" Version="9.0.0-dev.27" />
    <PackageVersion Include="Uno.Wasm.Bootstrap.Server" Version="9.0.0-dev.27" />
    <PackageVersion Include="Uno.WinUI" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.Lottie" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.DevServer" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.Skia.Gtk" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.WebAssembly" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.UITest.Helpers" Version="1.1.0-dev.70" />
    <PackageVersion Include="Xamarin.UITest" Version="4.3.3" />
  </ItemGroup>
</Project>

### Affected platforms

_No response_

### IDE

Visual Studio 2022

### IDE version

17.8.5

### Relevant plugins

Uno Platform 5.1.22.24

### Anything else we need to know?

I walking through your examples.  I start with using your wizard to create projects. I am testing some of Uno features the systems use of MVUX for navigation. But wanted to see how to use it with the NavigationView tools. I'm looking to see the extent use cases that allow for freestyle navigation.

Thanks so much,
Mark Allan

### PR DESCRIPTION
**Pull request by eriklimakc**
GitHub Issue (If applicable): closes #2159

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## What is the new behavior?

New page "How To: Define Regions"

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested with docfx
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/Learn/Navigation/HowTo-Regions.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+   - **Values**: `Visibility`.
+   - **Example**: `uen:Region.Navigator="Visibility"`
+
+1. **`Region.Name`**:
+   - **Description**: Assigns a name to the region, useful for identifying and referencing the region in code.
+   - **Example**: `uen:Region.Name="Main"`
+
+## How to use Regions
+
+Regions are particularly useful with navigational controls, such as `NavigationView` and `TabBar` for showing or hiding, and opening or closing content and pages. In these cases, it's recommendable to have a `Grid` element as the parent of the navigational control, that we will call as "Parent Grid" in this guide. Additionally, it is essential to define a `Grid` that will hold the content to be displayed based on the selected `NavigationViewItem` or `TabBarItem`, and we will call it a "Content Grid".
```
Comment (agneszitte) [inline doc/Learn/Navigation/HowTo-Regions.md:31]:
```suggestion
Regions are particularly useful with navigational controls, such as `NavigationView` and `TabBar` for showing or hiding, and opening or closing content and pages. In these cases, it's recommendable to have a `Grid` element as the parent of the navigational control, which we will call as "Parent Grid" in this guide. Additionally, it is essential to define a `Grid` that will hold the content to be displayed based on the selected `NavigationViewItem` or `TabBarItem`, and we will call it a "Content Grid".
```

File: doc/Learn/Navigation/HowTo-Regions.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+<Grid uen:Region.Attached="True">
+   <Grid uen:Region.Attached="True"
++        uen:Region.Navigator="Visibility" />
+   <utu:TabBar uen:Region.Attached="True">
+      <!-- Items -->
+   </utu:TabBar>
+</Grid>
+```
+
+The next important thing to do is to set the region names to associate the navigational control items to the content to be displayed. But before we give an example, we need to consider that there are two ways of setting the content in the Content `Grid`:
```
Comment (agneszitte) [inline doc/Learn/Navigation/HowTo-Regions.md:61]:
```suggestion
The next important thing to do is to set the region names to associate the navigational control items with the content to be displayed. But before we give an example, we need to consider that there are two ways of setting the content in the Content `Grid`:
```

File: doc/Learn/Navigation/HowTo-Regions.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            <TextBlock HorizontalAlignment="Center"
+                     VerticalAlignment="Center"
+                     FontSize="24"
+                     Text="Three" />
+         </StackPanel>
+      </Grid>
+   </Grid>
+   ```
+
+   For this case we can then associate the navigational control items with the contents by adding same region names:
```
Comment (agneszitte) [inline doc/Learn/Navigation/HowTo-Regions.md:91]:
```suggestion
   For this case, we can then associate the navigational control items with the contents by adding the same region names:
```

File: doc/Learn/Navigation/HowTo-Regions.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+- [How-To: Navigate using a TabBar](xref:Uno.Extensions.Navigation.Advanced.TabBar)
+- [How-To: Navigate using a NavigationView](xref:Uno.Extensions.Navigation.Advanced.NavigationView)
+
+## Navigators
+
+In *Uno.Extensions* navigation begins when a `Frame` is injected into the `ShellView`. This `Frame` acts as the main container for navigating between pages such as `WelcomePage`, `LoginPage`, and `MainPage`. The `Frame` has an associated navigator that handles the transitions between these pages.
+
+When there are nested views or pages, such as tabs defined through "Regions" within a `MainPage` for example, a second `Frame` is injected in the provided container to render these tabs content. This second `Frame` also has its own navigator, which manages navigation within the specific region of the tabs. Therefore, if a new page is opened from one of the tabs, this navigation is handled by the navigator associated with that specific region. This setup allows for hierarchical and organized navigation, where each region or tab maintains independent control over its page transitions, ensuring smooth and structured navigation within the application.
```
Comment (agneszitte) [inline doc/Learn/Navigation/HowTo-Regions.md:135]:
```suggestion
When there are nested views or pages, such as tabs defined through "Regions" within a `MainPage` for example, a second `Frame` is injected in the provided container to render these tabs' content. This second `Frame` also has its own navigator, which manages navigation within the specific region of the tabs. Therefore, if a new page is opened from one of the tabs, this navigation is handled by the navigator associated with that specific region. This setup allows for hierarchical and organized navigation, where each region or tab maintains independent control over its page transitions, ensuring smooth and structured navigation within the application.
```

**Comment by agneszitte** [comment]
@eriklimakc this doc does not have to wait for the next extensions stable package right?

**Comment by eriklimakc** [comment]
> @eriklimakc this doc does not have to wait for the next extensions stable package right?

@agneszitte No, this doesn't need to wait. This doc is for existing features.

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/4.2

File: doc/Learn/Navigation/HowTo-Regions.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+## Properties in the Region Class
+
+1. **`Region.Attached`**:
+   - **Description**: This property is used to mark a sector of the view as a region. It links the navigational control with the container provided to render content.
+   - **Example**: `<Grid uen:Region.Attached="True">`
+
+1. **`Region.Navigator`**:
+   - **Description**: Defines the navigation behavior for the region, determining how the region responds to navigation changes.
+   - **Values**: `Visibility`.
```
Comment (kazo0) [inline doc/Learn/Navigation/HowTo-Regions.md:22]: Is Visibility the only named navigator that we can put here? We can't force other things like a FrameNavigator?

File: doc/Learn/Navigation/HowTo-Regions.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+   - **Example**: `uen:Region.Navigator="Visibility"`
+
+1. **`Region.Name`**:
+   - **Description**: Assigns a name to the region, useful for identifying and referencing the region in code.
+   - **Example**: `uen:Region.Name="Main"`
+
+## How to use Regions
+
+Regions are particularly useful with navigational controls, such as `NavigationView` and `TabBar` for showing or hiding, and opening or closing content and pages. In these cases, it's recommendable to have a `Grid` element as the parent of the navigational control, which we will call as "Parent Grid" in this guide. Additionally, it is essential to define a `Grid` that will hold the content to be displayed based on the selected `NavigationViewItem` or `TabBarItem`, and we will call it a "Content Grid".
```
Comment (kazo0) [inline doc/Learn/Navigation/HowTo-Regions.md:31]:
```suggestion
Regions are particularly useful with navigational controls, such as `NavigationView` and `TabBar` for showing or hiding, and opening or closing content and pages. In these cases, it's recommended to have a `Grid` element as the parent of the navigational control, which we will refer to as the "Parent Grid" in this guide. Additionally, it is essential to define a `Grid` that will hold the content to be displayed based on the selected `NavigationViewItem` or `TabBarItem`, and we will call it a "Content Grid".
```

**Review by kazo0 — APPROVED** [review]

**Pull request merged**